src/read.rs

```diff
@@ -33,8 +34,19 @@ mod ffi {
     pub const S_IFREG: u32 = 0o0100000;
 }
 
+/// Extract immutable data from `ZipArchive` to make it cheap to clone
+#[derive(Debug)]
+struct Shared {
+    files: Vec<ZipFileData>,
+    names_map: HashMap<String, usize>,
+    offset: u64,
+    comment: Vec<u8>,
+}
+
 /// ZIP archive reader
 ///
+/// This type is cheap to clone if this is the case for the reader it uses.
```

Comment:
Please add that the performance of this may regress if you can ^^ We can merge either way